src/tests/ftest/vmd/fault_reintegration.py

```diff
@@ -159,10 +176,16 @@ def run_ior_with_nvme_fault(self, oclass=None):
             "Marking the {} device as faulty and verifying it is 'EVICTED' and its "
             "LED is 'ON'".format(test_dev))
         if not self.check_result(self.set_device_faulty(test_dev), "EVICTED", "ON"):
+            # Ensure the faulty device is reset in tearDown
+            self.register_cleanup(self.reset_fault_device, device=test_dev)
```

Comment:
Always register the cleanup action:
```suggestion
        check = self.check_result(self.set_device_faulty(test_dev), "EVICTED", "ON")

        # Ensure the faulty device is reset in tearDown
        self.register_cleanup(self.reset_fault_device, device=test_dev)

        if not check:
```